app/models/profile.rb

```diff
@@ -37,4 +32,26 @@ def self.static_fields
   def clear!
     update(data: {})
   end
+
+  # Lazily add accessors for profile fields on first use
+  def method_missing(method_name, *args, **kwargs, &block)
+    match = method_name.match(ATTRIBUTE_NAME_REGEX)
+    field = ProfileField.find_by(attribute_name: match[:attribute_name])
+    super unless field
+
+    self.class.instance_eval do
+      store_attribute :data, field.attribute_name.to_sym, field.type
+    end
+    public_send(method_name, *args, **kwargs, &block)
+  end
+
+  # Defining this is not only a good practice in general, it's also necessary
+  # for `update` to work since the `_assign_attribute` helper it uses performs
+  # an explicit `responds_to?` check.
+  def respond_to_missing?(method_name, include_private = false)
+    match = method_name.match(ATTRIBUTE_NAME_REGEX)
+    return true if match[:attribute_name].in?(self.class.attributes)
```

Comment:
this is one place we call `Profile.attributes` (and request data from the cached values) - I was reading this a little carefully trying to figure out if we were replacing a in-process cache (within the Profile class object) with a redis cache, and how often we would be making a network request to fetch.

Unless there's something calling `Profile.attributes` elsewhere, and frequently, this is probably a non-issue.

As a test I'll add logging just to get a feel for the frequency of that, but on the surface it looks like there's only this location in `:respond_to_missing?` for instances.